src/coap_main.c

```diff
@@ -365,30 +365,38 @@ CoAP_Result_t _rom CoAP_handleDisconnectEvt(uint32_t transport_ctx) {
 	CoAP_Result_t res = CoAP_FindObserverAndResourceByTransportCtx(transport_ctx,
 																   &pObserver,
 																   &pRes);
-	if (res != COAP_OK) {
-		return res;
-	}
+	if (res == COAP_OK) {
+		/* Notify resource about unlinking of the observer. */
+		pRes->ObserverInfo(pObserver, false, pRes, pObserver->Ep.session);
 
-	/* Notify resource about unlinking of the observer. */
-	pRes->ObserverInfo(pObserver, false, pRes, pObserver->Ep.session);
+		/* Delete/abort any pending notification interaction. */
+		res = CoAP_RemoveObserverInteractions(pObserver);
+		if (res != COAP_OK) {
+			return res;
+		}
 
-	/* Delete/abort any pending notification interaction. */
-	res = CoAP_RemoveObserverInteractions(pObserver);
-	if (res != COAP_OK) {
-		return res;
+		/* Remove observer from the resource. */
+		CoAP_RemoveObserverFromResource(&pRes->pListObservers,
+											pObserver->socketHandle,
+											&pObserver->Ep,
+											pObserver->Token);
 	}
-
-	/* Remove observer from the resource. */
-	res = CoAP_RemoveObserverFromResource(&pRes->pListObservers,
-										  pObserver->socketHandle,
-										  &pObserver->Ep,
-										  pObserver->Token);
-	if (res != COAP_REMOVED) {
-		return res;
+	
+	CoAP_Interaction_t *pIA = CoAP.pInteractions;
+	while ( NULL != pIA)
```

Comment:
I think that formatting should be (to be consistent with code in this repo):
```suggestion
	while (NULL != pIA)
```